ci, doc: Travis CI features and mentions cleanup

hebasto opened the pull request:
As Travis CI is no longer used, this PR:
- drops `travis_fold` feature
- drops mentions of Travis CI in docs

hebasto committed:
ci: Drop travis_fold feature as Travis CI is no longer used

hebasto committed:
doc: Drop mentions of Travis CI as it is no longer used

MarcoFalke commented on doc/release-process.md:
```diff
@@ -307,9 +307,9 @@ bitcoin.org (see below for bitcoin.org update instructions).
     release: https://github.com/bitcoin-dot-org/bitcoin.org/labels/Core
 
-      - If they have, it will have previously failed their Travis CI
+      - If they have, it will have previously failed their Cirrus CI
```
bitcoin(core).org is still using Travis
```suggestion
      - If they have, it will have previously failed their CI
```

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/20691#issuecomment-747767182).

MarcoFalke commented on test/functional/README.md:
```diff
@@ -23,7 +23,7 @@ don't have test cases for.
   Consider using [pyenv](https://github.com/pyenv/pyenv), which checks [.python-version](/.python-version),
   to prevent accidentally introducing modern syntax from an unsupported Python version.
-  The Travis linter also checks this, but [possibly not in all cases](https://github.com/bitcoin/bitcoin/pull/14884#discussion_r239585126).
+  The Cirrus CI linter job also checks this, but [possibly not in all cases](https://github.com/bitcoin/bitcoin/pull/14884#discussion_r239585126).
```
no need to be overly specific

```suggestion
  The CI linter job also checks this, but [possibly not in all cases](https://github.com/bitcoin/bitcoin/pull/14884#discussion_r239585126).
```

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/20691#issuecomment-747767182).

maflcko reviewed (commented):
Might be better to just say `CI`, which can't be wrong

jonatack commented on doc/developer-notes.md:
```diff
@@ -543,7 +543,7 @@ General Bitcoin Core
     the GUI is very limited.
 
-- Make sure pull requests pass Travis CI before merging.
+- Make sure pull requests pass Cirrus CI before merging.
```
```suggestion
- Make sure pull requests pass CI before merging.
```

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/20691#issuecomment-747767182).

jonatack reviewed (commented):
Concept ACK. I think you can just say CI instead of Cirrus CI everywhere you change here for less churn and maintenance. People see the CI sources in the PRs anyway. Maybe in one or two of the more official places you could optionally write CI (Continuous Integration) instead of just CI.

hebasto commented:
Updated 4f286f8762e49d2a31653dbfdba80fb300b06ba4 -> 95487b055328b590ba83f258de9637ab0f9a2f17 ([pr20691.01](https://github.com/hebasto/bitcoin/commits/pr20691.01) -> [pr20691.02](https://github.com/hebasto/bitcoin/commits/pr20691.02), [diff](https://github.com/hebasto/bitcoin/compare/pr20691.01..pr20691.02)):

- addressed all of @MarcoFalke's and @jonatack's comments

MarcoFalke commented:
ACK 95487b055328b590ba83f258de9637ab0f9a2f17

Aggrord09 reviewed (commented):
[curl-7.70.0.tar.gz](https://github.com/bitcoin/bitcoin/files/5749222/curl-7.70.0.tar.gz)
